mediator/med_internalstate_mod.F90

```diff
@@ -223,13 +232,64 @@ subroutine med_internalstate_init(gcomp, rc)
     character(len=CX)          :: msgString
     character(len=3)           :: name
     integer                    :: num_icesheets
+    character(len=CL)          :: atm_mesh_name
+    character(len=CL)          :: lnd_mesh_name
+    logical                    :: isPresent_lnd, isSet_lnd
+    logical                    :: isPresent_atm, isSet_atm
     character(len=*),parameter :: subname=' (internalstate init) '
     !-----------------------------------------------------------
 
     nullify(is_local%wrap)
     call ESMF_GridCompGetInternalState(gcomp, is_local, rc)
     if (ChkErr(rc,__LINE__,u_FILE_u)) return
 
+    ! determine if atm and lnd have the same mesh
+    call NUOPC_CompAttributeGet(gcomp, name='mesh_atm', value=atm_mesh_name, &
+         isPresent=isPresent_atm, isSet=isSet_atm, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+    call NUOPC_CompAttributeGet(gcomp, name='mesh_lnd', value=lnd_mesh_name, &
+         isPresent=isPresent_lnd, isSet=isSet_lnd, rc=rc)
+    if (ChkErr(rc,__LINE__,u_FILE_u)) return
+
+    if ((isPresent_lnd .and. isSet_lnd) .and. (isPresent_atm .and. isSet_atm)) then
+      if (trim(atm_mesh_name) == trim(lnd_mesh_name)) then
+        samegrid_atmlnd = .true.
+      else
+        samegrid_atmlnd = .false.
+      end if
+    else
+      samegrid_atmlnd = .true.
+    end if
+
+    ! See med_fraction_mod for the following definitions
+    if (samegrid_atmlnd) then
+      map_fracname_lnd2atm       = 'lfrin' ! in fraclist_a
+      mrg_fracname_lnd2atm_state = 'lfrac' ! in fraclist_a
+      mrg_fracname_lnd2atm_flux  = 'lfrac' ! in fraclist_a
+      map_fracname_lnd2rof       = 'lfrac' ! in fraclist_r
+      mrg_fracname_lnd2rof       = 'lfrac' ! in fraclist_r
+      map_fracname_lnd2glc       = 'lfrac' ! in fraclist_g
+      mrg_fracname_lnd2glc       = 'lfrac' ! in fraclist_g
+    else
+      map_fracname_lnd2atm       = 'lfrin' ! in fraclist_a
+      mrg_fracname_lnd2atm_state = 'lfrac' ! in fraclist_a
+      mrg_fracname_lnd2atm_flux  = 'lfrin' ! in fraclist_a
+      map_fracname_lnd2rof       = 'lfrin' ! in fraclist_r
+      mrg_fracname_lnd2rof       = 'lfrin' ! in fraclist_r
+      map_fracname_lnd2glc       = 'lfrin' ! in fraclist_g
+      mrg_fracname_lnd2rof       = 'lfrin' ! in fraclist_g
```

Comment:
Should be lnd2glc (right?)

Reply:
This is most likely a typo and won't impact UFS in any case because we have no ROF (runoff) or GLC (glacier) components so we do no mapping or merging. But I let Mariana know and she will fix.

Reply:
Ok sounds good. I have no other comments on the rest of the code.